dev-db/tarantool: Failed to emerge dev-db/tarantool-2.11.0

I tried to upgrade my Tarantool atom while emerging the world today. The full pack of the artefacts is uploaded to transfer.sh:
* [build.log](https://transfer.sh/c5FCRUwhGz/tarantool-2.11.0-build.log)
* [environment](https://transfer.sh/wvQOHAbpah/tarantool-2.11.0-environment)
* [emerge --info '=dev-db/tarantool-2.11.0::tarantool'](https://transfer.sh/wvQOHAbpah/tarantool-2.11.0-environment)
* [emerge -pqv '=dev-db/tarantool-2.11.0::tarantool'](https://transfer.sh/6JEeUT6Que/tarantool-2.11.0-emerge--pqv)

You can also find the relevant part below:
```
<snipped>
[ 56%] Building C object src/box/CMakeFiles/xlog.dir/xlog.c.o
cd /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/src/box && /usr/bin/x86_64-pc-linux-gnu-gcc -DCORO_ASM -DDT_NAMESPACE=tnt_ -DLUAJIT_SMART_STRINGS=1 -DNVALGRIND=1 -DSQL_DEBUG=1 -DSQL_TEST=1 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D__STDC_LIMIT_MACROS=1 -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/src -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/lib -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/lib/small/include -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/lib/small/third_party -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/lib/core -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/xxHash -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/luajit/src -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/third_party -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/coro -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/third_party/luajit/src -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/lib/msgpuck -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/third_party/decNumber -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/libutil_freebsd -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/src/box/sql -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/src/box -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/sql -I/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/c-dt -isystem /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0_build/third_party/libunwind/include -isystem /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/third_party/libunwind/include  -O2 -pipe -march=sandybridge -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -fopenmp -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type -Werror  -ggdb -O0 -MD -MT src/box/CMakeFiles/xlog.dir/xlog.c.o -MF CMakeFiles/xlog.dir/xlog.c.o.d -o CMakeFiles/xlog.dir/xlog.c.o -c /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c
/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c: In function 'xlog_tx_write_zstd':
/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:1162:9: error: 'ZSTD_compressBegin' is deprecated [-Werror=deprecated-declarations]
 1162 |         ZSTD_compressBegin(log->zctx, 3);
      |         ^~~~~~~~~~~~~~~~~~
In file included from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.h:40,
                 from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:31:
/usr/include/zstd.h:2849:27: note: declared here
 2849 | ZSTDLIB_STATIC_API size_t ZSTD_compressBegin(ZSTD_CCtx* cctx, int compressionLevel);
      |                           ^~~~~~~~~~~~~~~~~~
/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:1182:25: error: 'ZSTD_compressEnd' is deprecated [-Werror=deprecated-declarations]
 1182 |                         fcompress = ZSTD_compressEnd;
      |                         ^~~~~~~~~
In file included from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.h:40,
                 from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:31:
/usr/include/zstd.h:2862:27: note: declared here
 2862 | ZSTDLIB_STATIC_API size_t ZSTD_compressEnd(ZSTD_CCtx* cctx, void* dst, size_t dstCapacity, const void* src, size_t srcSize);
      |                           ^~~~~~~~~~~~~~~~
/var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:1184:25: error: 'ZSTD_compressContinue' is deprecated [-Werror=deprecated-declarations]
 1184 |                         fcompress = ZSTD_compressContinue;
      |                         ^~~~~~~~~
In file included from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.h:40,
                 from /var/tmp/portage/dev-db/tarantool-2.11.0/work/tarantool-2.11.0/src/box/xlog.c:31:
/usr/include/zstd.h:2860:27: note: declared here
 2860 | ZSTDLIB_STATIC_API size_t ZSTD_compressContinue(ZSTD_CCtx* cctx, void* dst, size_t dstCapacity, const void* src, size_t srcSize);
      |                           ^~~~~~~~~~~~~~~~~~~~~
<snipped>
```

There might be some tweaks required for my build environment, but anyway, the build failed.

--- a/dev-db/tarantool/tarantool-9999.ebuild
+++ b/dev-db/tarantool/tarantool-9999.ebuild
@@ -164,6 +164,12 @@ src_configure() {
 		export CMAKE_BUILD_TYPE=RelWithDebInfo
 	fi
 
+	# https://github.com/tarantool/gentoo-overlay/issues/73
+	if use system-zstd; then
+		CFLAGS="$CFLAGS -Wno-deprecated-declarations"
+		CXXFLAGS="$CXXFLAGS -Wno-deprecated-declarations"
+	fi
+
 	local mycmakeargs=(
 		-DENABLE_BACKTRACE="$(usex backtrace)"
 		-DENABLE_SSE2="$(usex cpu_flags_x86_sse2)"

